### DISCUSSION TITLE
我想用scoop下载GitHub上的安装文件都走代理，如何设置

### DISCUSSION THREAD

**Discussion by Niefee** [clash-verge-rev/clash-verge-rev#1337] [General]
![image](https://github.com/clash-verge-rev/clash-verge-rev/assets/10347495/21d810cf-93a2-4711-846c-2577da605840)
这样设置，然后点击添加前置规则，好像不起效，连接面板看不到相关日志

一般下载地址是`https://github.com/pnpm/pnpm/releases/download/v9.4.0/pnpm-win-x64.exe`然后会重定向成 

```url
https://objects.githubusercontent.com/github-production-release-asset-2e65be/50565430/a818e4b5-909e-43eb-a55c-4aeff866eab4?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=releaseassetproduction%2F20240705%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20240705T081354Z&X-Amz-Expires=300&X-Amz-Signature=8be872542b259762b49b2e89589ec584fb86b57860ba19ddfb4d38d834a90f7a&X-Amz-SignedHeaders=host&actor_id=10347495&key_id=0&repo_id=50565430&response-content-disposition=attachment%3B%20filename%3Dpnpm-win-x64.exe&response-content-type=application%2Foctet-stream
```

**Comment by asdasdqqq** [comment]
可以让scoop直接走代理 
```
scoop config proxy 127.0.0.1:7897
```
或者TUN模式

**Comment by Charlotte4177** [comment]
> ![image](https://private-user-images.githubusercontent.com/10347495/346054927-21d810cf-93a2-4711-846c-2577da605840.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjcxNzUzODAsIm5iZiI6MTcyNzE3NTA4MCwicGF0aCI6Ii8xMDM0NzQ5NS8zNDYwNTQ5MjctMjFkODEwY2YtOTNhMi00NzExLTg0NmMtMjU3N2RhNjA1ODQwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MjQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTI0VDEwNTEyMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWJjYzZlNzU0YzFmZDZiMjI2ZTVmZGVlNTVmMDAxYmRhNDlmNzAzYTJhYmJlNTE3MDZlNjc1YmEyYjc1YTI1OGQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.EQ-oesv6HLJFEsN9B7ip41d3N3fB0brtN4365YYQIaY) 这样设置，然后点击添加前置规则，好像不起效，连接面板看不到相关日志
> 
> 一般下载地址是`https://github.com/pnpm/pnpm/releases/download/v9.4.0/pnpm-win-x64.exe`然后会重定向成
> 
> ```ini
> https://objects.githubusercontent.com/github-production-release-asset-2e65be/50565430/a818e4b5-909e-43eb-a55c-4aeff866eab4?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=releaseassetproduction%2F20240705%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20240705T081354Z&X-Amz-Expires=300&X-Amz-Signature=8be872542b259762b49b2e89589ec584fb86b57860ba19ddfb4d38d834a90f7a&X-Amz-SignedHeaders=host&actor_id=10347495&key_id=0&repo_id=50565430&response-content-disposition=attachment%3B%20filename%3Dpnpm-win-x64.exe&response-content-type=application%2Foctet-stream
> ```

下载链接的重定向是由 GitHub 和其存储服务（如 AWS S3）管理的，通常无法避免。不过，你可以尝试以下方法直接下载：

1. **使用下载管理器**：一些下载管理器可以处理重定向链接，帮助你直接下载文件。
2. **手动复制链接**：在浏览器中打开重定向后的链接，手动复制最终的下载地址，然后粘贴到下载管理器或浏览器的下载栏中。

如果你需要进一步的帮助或有其他问题，请告诉我！

**Comment by TomZhang1019** [comment]
直接TUN模式, 别说scoop走代理, WSL和虚拟机不用配置, 也直接就走代理

**Comment by zweie** [comment]
先scoop设置代理，命令就是上面那个。然后给git设置代理，命令好像是这个来着 `git config --global http.proxy http://user:password@server:port`

规则你直接一个geosite不就好了？
